File: frontend/types/observation.ts
export interface Observation {
  id: string
  claim_id: string
  author_user_id: string | null
  comment: string
  is_internal: boolean
  tenant_id: string
  created_at: string
  updated_at: string | null
}

export interface ObservationListResponse {
  items: Observation[]
  total: number
  page: number
  limit: number
}

export interface ObservationCreatePayload {
  comment: string
  is_internal: boolean
}

File: backend/app/schemas/observation.py
"""Pydantic schemas for observations."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ObservationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    claim_id: UUID
    author_user_id: UUID | None
    comment: str
    is_internal: bool
    tenant_id: UUID
    created_at: datetime
    updated_at: datetime | None


class ObservationCreate(BaseModel):
    comment: str = Field(min_length=1, max_length=5000)
    is_internal: bool = False


class ObservationListResponse(BaseModel):
    items: list[ObservationOut]
    total: int
    page: int
    limit: int
